### PR TITLE
Manage Helm chart CRDs outside of Helm

### PR DESCRIPTION
**Pull request by phillebaba**
I have on purpose skipped Linkerd as I will solve it in a separate PR when I bump the version.

### CODE REVIEW

File: modules/kubernetes/aad-pod-identity/main.tf [hunk 1]
```diff
@@ -29,8 +33,33 @@ resource "kubernetes_namespace" "this" {
...
+  }
+  server_side_apply = true
+  apply_only        = true
+  yaml_body         = each.value
+}
+
 #tf-latest-version:ignore
 resource "helm_release" "aad_pod_identity" {
+  depends_on = [kubectl_manifest.aad_pod_identity]
+
```
Comment (NissesSenap) [inline modules/kubernetes/aad-pod-identity/main.tf:62]: Don't we need the skip_crds on aad-pod-identity as well?

File: modules/kubernetes/datadog/main.tf [hunk 2]
```diff
@@ -44,7 +48,32 @@ resource "kubernetes_namespace" "this" {
   }
 }
 
+data "helm_template" "datadog_operator" {
+  repository   = "https://helm.datadoghq.com"
+  chart        = "datadog-operator"
+  name         = "datadog-operator"
+  version      = "0.7.0"
+  api_versions = ["apiextensions.k8s.io/v1/CustomResourceDefinition"]
```
Comment (NissesSenap) [inline modules/kubernetes/datadog/main.tf:56]: Such a nice workaround :)

File: modules/kubernetes/opa-gatekeeper/main.tf [hunk 3]
```diff
@@ -118,13 +122,39 @@ resource "kubernetes_namespace" "this" {
...
+  name         = "gatekeeper"
+  version      = "3.7.1"
+  include_crds = true
+}
+
+data "kubectl_file_documents" "gatekeeper" {
+  content = data.helm_template.gatekeeper.manifest
+}
+
+resource "kubectl_manifest" "gatekeeper" {
```
Comment (NissesSenap) [inline modules/kubernetes/opa-gatekeeper/main.tf:137]: Don't gatekeeper manage the crd on it's own?

File: modules/kubernetes/cert-manager/main.tf [hunk 4]
```diff
@@ -34,13 +38,42 @@ resource "kubernetes_namespace" "this" {
...
+    name  = "installCRDs"
+    value = "true"
+  }
+}
+
+data "kubectl_file_documents" "cert_manager" {
+  content = data.helm_template.cert_manager.manifest
+}
+
+resource "kubectl_manifest" "cert_manager" {
```
Comment (NissesSenap) [inline modules/kubernetes/cert-manager/main.tf:56]: Don't cert-manager manage the crd on it's own?

**Review by NissesSenap — APPROVED** [review]
Minor comments, if you don't think it's a problem go for it :)

File: modules/kubernetes/opa-gatekeeper/main.tf (hunk 3 above)
Comment (phillebaba) [reply modules/kubernetes/opa-gatekeeper/main.tf:137]: I want to manage the CRDs in the same way for all charts, even if they have their own method to update CRDs

**Comment by NissesSenap** [comment]
Output from a clean cluster:

```.hcl
│ Error: Invalid for_each argument
│ 
│   on .terraform/modules/aks1_core/modules/kubernetes/aad-pod-identity/main.tf line 49, in resource "kubectl_manifest" "aad_pod_identity":
│   49:   for_each = {
│   50:     for k, v in data.kubectl_file_documents.aad_pod_identity.manifests :
│   51:     k => v
│   52:     if can(regex("^/apis/apiextensions.k8s.io/v1/customresourcedefinitions/", k))
│   53:   }
│     ├────────────────
│     │ data.kubectl_file_documents.aad_pod_identity.manifests is a map of string, known only after apply
│ 
│ The "for_each" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work
│ around this, use the -target argument to first apply only the resources that the for_each depends on.
...
etc
``` 

I will give it a try with depends on.

**Review by NissesSenap — APPROVED** [review]
LGTM, great job!

**Pull request merged**